crates/pop-wallet-integration/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+use axum::{
+	routing::{get, post},
+	Router,
+};
+use serde::Serialize;
+use std::{path::PathBuf, sync::Arc};
+use tokio::sync::{oneshot, Mutex};
+use tower_http::services::ServeDir;
+
+/// Data to be sent to frontend for signing.
+#[derive(Serialize)]
+pub struct Data {
+	chain_rpc: String,
+	data_type: DataType,
+}
+
+/// The type of transaction with specific data for signing (contract or parachain).
+#[derive(Serialize)]
+pub enum DataType {
+	/// Parachain call, where Vec<u8> is the encoded call data.
+	Parachain(Vec<u8>),
+	Contract(ContractArgs),
```

Comment:
We should be able to benefit from the work done around `pop call` and `pop up` to intake the payloads

Reply:
no longer needed